src/wallet/wallet.cpp

```diff
@@ -4303,7 +4303,7 @@ int CWallet::ScanForWalletTransactions(
 
         // no need to read and scan block, if block was created before
         // our wallet birthday (as adjusted for block time variability)
-        while (pindex && nTimeFirstKey && pindex->GetBlockTime() < nTimeFirstKey - TIMESTAMP_WINDOW) {
+        while (chainActive.Next(pindex) != NULL && nTimeFirstKey && pindex->GetBlockTime() < nTimeFirstKey - TIMESTAMP_WINDOW) {
```

Comment:
This change is not quite the same as the change upstream made that I referenced in https://github.com/zcash/zcash/issues/5712#issuecomment-1076775990, but AFAICT the difference is that upstream the result would be rescanning from genesis, whereas here the result would be not rescanning at all (as if we reach this condition we've reached chain tip). So I'm fine with the fix here as-is, but I'm also happy to eventually backport upstream's changes to overwrite this.